src/RootContainer.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import { queryData, getWidgets } from './query';
+import WidgetList from './components/WidgetList';
+import BarChart from './components/BarChart';
+import FilterPanel from './components/FilterPanel';
+import Loading from './components/Loading';
+
+const RootContainer = ({ serviceUrl, entity }) => {
+	const [filterOptions, setFilterOptions] = useState({
+		maxp: 0.05,
+		processFilter: 'biological_process',
+		correction: 'Holm-Bonferroni',
+		limitResults: 20
+	});
+	const [data, setData] = useState([]);
+	const [widgetList, setWidgetList] = useState([]);
+	const [graphData, setGraphData] = useState([]);
+	const [selectedWidget, setSelectedWidget] = useState({});
+	const [loading, setLoading] = useState(false);
+	const [chartLoading, setChartLoading] = useState(false);
+	const [maxResultLimit, setMaxLimit] = useState(0);
+
+	useEffect(() => {
+		setLoading(true);
+		getWidgets({ serviceUrl })
+			.then(res => {
+				const enrichmentWidgets = res.filter(
+					w =>
+						w.widgetType === 'enrichment' &&
+						w.targets.indexOf(entity.class) !== -1
```

Comment:
Tip: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes is pretty nice!